tools/buildsteps/windows/x64-uwp/default.bat

```diff
@@ -0,0 +1,23 @@
+@echo OFF
+
+pushd %~dp0\..
+call .helpers\default.bat
+call vswhere.bat x64 store
+if ERRORLEVEL 1 (
+	echo ERROR! Something went wrong when calling vswhere.bat
+	popd
+	exit /B 1
+)
+popd
+
+if "%Configuration%"=="" (
```

Comment:
How do we specify RelWithDebInfo build? Setting an environment variable is not intuitive and not something that's normal in Windows.

We need both RelWithDebInfo and Debug libraries for a developer setup but the toolchain file is per configuration?

Reply:
`make-target-depends.bat` need to generate both `RelWithDebInfo` and `Debug` version of static libs, same as we provide in the prepackaged depends today.

Setting environment variables to control things is not intuitive and we should not rely on that.

Reply:
No, you decided for which configuration to build, and then build depends and Kodi with that configuration.

Reply:
no you generate a project and then in VS toggle between debug and RelWithDebInfo depending on what you need. It's not acceptable to have to keep two solutions depending on config.

for Jenkins it makes sense to only build RelWithDebInfo but for a developer setup both are required in the same solution.

Reply:
It is not uncommon in cross-platform to generate different solutions for different configurations. for now maybe just leave it like @Rechi suggests

Reply:
posix platforms work just like this as well. You specify release/debug/... at configure time and then build everything.

Reply:
It's a huge step backwards from what we currently have on windows so it needs to be solved before we merge this.

Reply:
This will not happen, because it makes the whole process error prone and doubles the build time of target depends for no reason.

Usually one only uses the debug configuration, unless there is a very specific bug to fix which only happens in release configuration.

If two configurations get installed into the same prefix (even with different library names, which won't be picked up anyway in most cases) one variant of the include files will be overwritten and there is no guarantee that include files installed by a debug configuration are compatible with building Kodi in release configuration or the other way around.

Reply:
Usually we only use RelWithDebInfo versions of dependencies because it's rare to debug the dependencies themselves and perf is better that way.

The libraries that requires both debug and release versions for Kodi to build need to be built in both configs. Also if one is working on the dependencies it needs to support building both versions from the same generated solution.

If we can't get feature parity with our current kodi-deps it's better to stay with that solution

Reply:
In general one can't mix Debug and Release/RelWithDebInfo configurations. This is not a limitation of Kodi, but of Microsoft, because `_ITERATOR_DEBUG_LEVEL` and `RuntimeLibrary` mismatch.

So for building Kodi with Debug configuration you need the libraries in Debug Configuration. In that case the additional time spent for additionally building target dependencies in another configuration is just wasted.

Why does the solution have to support multiple configurations if you are working on dependencies? The reason for disabling this in general is that people just wanting to build Kodi can't get anything wrong. If you need it for whatever reason the following diff will disable the safety check, but then you are on your own.

```diff
--- a/tools/windows/depends/target/Toolchain.cmake.in
+++ b/tools/windows/depends/target/Toolchain.cmake.in
@@ -1,5 +1,5 @@
 set(CMAKE_SYSTEM_NAME @CMAKE_SYSTEM_NAME@)
 set(CMAKE_SYSTEM_VERSION @CMAKE_SYSTEM_VERSION@)
-if(NOT CMAKE_PROJECT_NAME STREQUAL CMAKE_TRY_COMPILE)
+if(OFF NOT CMAKE_PROJECT_NAME STREQUAL CMAKE_TRY_COMPILE)
 	set(CMAKE_CONFIGURATION_TYPES @CMAKE_CONFIGURATION_TYPES@)
 	if(@CMAKE_CONFIGURATION_TYPES@ STREQUAL Release)
```

Reply:
Need to test that both configs work